x/crosschain/keeper/keeper_tss.go

```diff
@@ -36,22 +45,32 @@ func (k Keeper) RemoveTSS(ctx sdk.Context) {
 	store.Delete([]byte{0})
 }
 
-// GetAllTSS returns all tss information
-func (k Keeper) GetAllTSS(ctx sdk.Context) (list []types.TSS) {
-	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.TSSKey))
+// GetAllTSS returns all tss historical information from the store
+func (k Keeper) GetAllTSS(ctx sdk.Context) (list []*types.TSS) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.TSSHistoryKey))
 	iterator := sdk.KVStorePrefixIterator(store, []byte{})
-
 	defer iterator.Close()
-
 	for ; iterator.Valid(); iterator.Next() {
 		var val types.TSS
 		k.cdc.MustUnmarshal(iterator.Value(), &val)
-		list = append(list, val)
+		list = append(list, &val)
 	}
-
 	return
 }
 
+// GetPreviousTSS returns the previous tss information
+func (k Keeper) GetPreviousTSS(ctx sdk.Context) (val types.TSS, found bool) {
+	tssList := k.GetAllTSS(ctx)
+	if len(tssList) <= 2 {
```

Comment:
```suggestion
	if len(tssList) < 2 {
```
I believe you can have a previous tss if len==2